contracts/src/internal/HyperdriveBase.sol

```diff
@@ -682,6 +677,152 @@ abstract contract HyperdriveBase is IHyperdriveEvents, HyperdriveStorage {
         return (lpSharePrice, true);
     }
 
+    /// @dev Calculates the pool's solvency if a long is opened that brings the
+    ///      rate to 0%. This is the maximum possible long that can be opened on
+    ///      the YieldSpace curve.
+    /// @param _shareReserves The pool's share reserves.
+    /// @param _shareAdjustment The pool's share adjustment.
+    /// @param _bondReserves The pool's bond reserves.
+    /// @param _vaultSharePrice The vault share price.
+    /// @param _longExposure The pool's long exposure.
+    /// @param _checkpointExposure The pool's checkpoint exposure.
+    /// @return The solvency after opening the max long.
+    /// @return A flag indicating whether or not the calculation succeeded.
+    function _calculateSolvencyAfterMaxLongSafe(
```

Comment:
just curious why you moved this to HyperdriveBase from LPMath

Reply:
It ends up being cheaper because we ran into stack too deep issues. Within hyperdrive base, I can access the immutables without having to create new stack variables for them. I actually moved it into the library to try to fix the code size issue, but it made it worse and forced me to create a memory struct which would increase gas costs.

Reply:
okay ya. that makes sense